src/unix_term.rs

```diff
@@ -48,10 +48,16 @@ pub fn read_secure() -> io::Result<String> {
         }
     };
 
-    let mut termios = termios::Termios::from_fd(fd)?;
+    let mut termios = core::mem::MaybeUninit::uninit();
+    if unsafe { libc::tcgetattr(fd, termios.as_mut_ptr()) } != 0 {
+        return Err(io::Error::last_os_error());
```

Comment:
Can you please extract out this error returning into a function that returns `Result` and use the `?` operator more?